spec.html

```diff
@@ -531,114 +610,104 @@ <h1>%AsyncIterator.prototype%.constructor</h1>
-            1. IfAbruptCloseAsyncIterator(_iterated_, _selected_).
-            1. Set _selected_ to Await(_selected_).
-            1. IfAbruptCloseAsyncIterator(_iterated_, _selected_).
-            1. If ToBoolean(_selected_) is *true*, then
-              1. Set _lastValue_ to Yield(_value_).
-              1. IfAbruptCloseAsyncIterator(_iterated_, _lastValue_).
+          1. Let _closure_ be a new Abstract Closure with no parameters that captures _iterated_ and _filterer_ and performs the following steps when called:
+            1. Let _lastValue_ be *undefined*.
+            1. Repeat,
+              1. Let _next_ be ? Await(? IteratorNext(_value_, _lastValue_)).
+              1. If ? IteratorComplete(_next_) is *true*, return *undefined*.
+              1. Let _value_ be ? IteratorValue(_next_).
+              1. Let _selected_ be Call(_filterer_, *undefined*, &laquo; _value_ &raquo;).
+              1. IfAbruptCloseAsyncIterator(_iterated_, _selected_).
+              1. Set _selected_ to Await(_selected_).
+              1. IfAbruptCloseAsyncIterator(_iterated_, _selected_).
+              1. If ! ToBoolean(_selected_) is *true*, then
+                1. Set _lastValue_ to Yield(_value_).
+                1. IfAbruptCloseAsyncIterator(_iterated_, _lastValue_).
+          1. Return ? CreateBuiltinAsyncGeneratorInstance(_closure_, ~Async Iterator Helper~, %AsyncIteratorHelperPrototype%).
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-asynciteratorprototype.take">
         <h1>%AsyncIterator.prototype%.take ( _limit_ )</h1>
-        <p>%AsyncIterator.prototype%.take is a built-in async generator function which, when called, performs the following prelude steps:</p>
         <emu-alg>
           1. Let _iterated_ be ? GetIteratorDirect(*this* value).
           1. Let _remaining_ be ? ToInteger(_limit_).
           1. If _remaining_ &lt; 0, throw a *RangeError* exception.
-        </emu-alg>
-        <p>The body of %AsyncIterator.prototype%.take is composed of the following steps:</p>
-        <emu-alg>
-          1. Let _lastValue_ be *undefined*.
-          1. Repeat,
-            1. If _remaining_ is 0, then
-              1. Return ? Await(? AsyncIteratorClose(_iterated_, NormalCompletion(*undefined*))).
-            1. Set _remaining_ to _remaining_ - 1.
-            1. Let _next_ be ? Await(? IteratorNext(_iterated_, _lastValue_)).
-            1. If ? IteratorComplete(_next_) is *true*, return *undefined*.
-            1. Set _lastValue_ to Yield(? IteratorValue(_next_)).
-            1. IfAbruptCloseAsyncIterator(_iterated_, _lastValue_).
+          1. Let _closure_ be a new Abstract Closure with no parameters that captures _iterated_ and _remaining_ and performs the following steps when called:
+            1. Let _lastValue_ be *undefined*.
+            1. Repeat,
+              1. If _remaining_ is 0, then
+                1. Return ? Await(? AsyncIteratorClose(_iterated_, NormalCompletion(*undefined*))).
```

Comment:
does this need return + await? can it just be `Return`, since the closure is evaluated in the context of a promise?

Reply:
I think that Await is redundant, yeah